bin/libast_prereq.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/sh
+
+# This script is used for feature detection and setting up other configurations that are required to build libast
+set -x
+set -e
+
+# This script is run from an unspecified directory so we have to determine directory paths
+# http://mesonbuild.com/Reference-manual.html#run_command
+script_path=`realpath $0`
+bin_dir=`dirname $script_path`
+base_dir=`dirname $bin_dir`
+
+PATH=$bin_dir:$PATH
+
+# Legacy build system keeps files generated by iffe with different names under different directories
+# Each array listed below is processed separately and headers generated by iffe are kept in separate directories
+# in a way that is compatible with legacy build system
+iffe_tests=( 'api' 'aso' 'ccode' 'common' 'float' 'fs' 'iconv' 'lib' 'locale' 'mmap' 'ndbm' 'options' 'prog' 'sched' 'sfio' 'siglist' 'sizeof' 'spawn' 'standards' 'sys' 'sys_calls' 'time' 'tmlib' 'tty' 'tv' 'tvlib' 'vfork' 'vmalloc' 'wait')
+
+iffe_tests_2=( 'tv' 'tmx' )
+
+iffe_c_tests=('align.c' 'fcntl.c'  'limits.c'  'map.c'  'mode.c') 
+
+iffe_c_tests_2=('atdev.c' 'sfinit.c' 'signal.c')
+
+iffe_c_tests_3=('align.c')
+
+iffe_sh_tests=('preroot.sh'  'sig.sh')
+
+iffe_sh_tests_2=('param.sh' )
+
+iffe_sh_tests_3=('libpath.sh' )
+
+pushd $base_dir/src/lib/libast/features
+
+iffe -v -X ast -X std -c 'cc -D_BLD_DLL -D_BLD_ast    ' run param.sh
+mv $base_dir/src/lib/libast/features/FEATURE/param.sh $base_dir/src/lib/libast/features/FEATURE//param
+
+for iffe_test in ${iffe_tests[@]}; do
+    iffe -v -X ast -X std -c 'cc -D_BLD_DLL -D_BLD_ast    ' run $iffe_test
+
+    cp "$base_dir/src/lib/libast/features/FEATURE/$iffe_test" "$base_dir/src/lib/libast/features/ast_$iffe_test.h"
+done
+
+for iffe_test in ${iffe_tests_2[@]}; do
+    iffe -v -X ast -X std -c 'cc -D_BLD_DLL -D_BLD_ast    ' run $iffe_test
+    
+    cp "$base_dir/src/lib/libast/features/FEATURE/$iffe_test" "$base_dir/src/lib/libast/features/$iffe_test.h"
+done
+
+$bin_dir/conf -v $base_dir/src/lib/libast/comp/conf.tab cc -D_BLD_DLL -D_BLD_ast
+
+# iffe behaves strangely when I try to pass it a string made with double quotes
+# I pass this function as an argument to iffe while processing .c files
+function cc_fun {
+    cc -D_BLD_DLL -D_BLD_ast -I. -I.. -Icomp -I../comp -Iinclude -I../include -Istd -I../std -I../features  "$@"
```

Comment:
Wondering, whether there is a way to say `gcc` instead of `cc` (or honor the CC env var) ...